exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -956,21 +925,8 @@ private void spillAPartition(int part) {
       this.htables[part].outputKeys(currOutBatchIndex, this.outContainer, outStartIdxHolder.value, outNumRecordsHolder.value, numPendingOutput);
 
       // set the value count for outgoing batch value vectors
-      /* int i = 0; */
       for (VectorWrapper<?> v : outgoing) {
         v.getValueVector().getMutator().setValueCount(numOutputRecords);
-        /*
```

Comment:
We don't have any facility to take a peek at actual data for debugging. While this commented code looks ugly, it allows for easy enabling to look at data, if needed (or an example of how to do so, for elsewhere).

Reply:
If this logic is really critical for debugging we should have utility methods to help with it so that others can benefit from the code. Maybe we can brainstorm about what methods would be helpful for debugging and start building a utility library. In the meantime if you need this code to debug you can easily look into the history and copy and paste it from there.

Reply:
See the `RowSet` classes. Chunhui and others have successfully used these to print bathes for debugging. Roughly:

```
RowSet rs = DirectRowSet.fromContainer(container);
rs.print();
```

This can be modified to take a row range (to avoid, say, printing 64K rows.) There are versions to do the printing using an SV2 or SV4.

In the code I've not yet posted is a utility to print the contents of an offset vector. Very handy when your offsets get messed up.

Reply:
Thanks!